### PR TITLE
Use the path of libmcwamp to dlopen libmcwamp_hsa

### PR DESCRIPTION
**Pull request by scchan**

### CODE REVIEW

File: lib/mcwamp.cpp [hunk 1]
```diff
@@ -428,6 +433,22 @@ KalmarContext *getContext() {
...
+        if (pos != std::string::npos) {
+          p.erase(p.begin()+pos, p.end());
+          CLAMP::library_path = std::move(p);
+          return 1;
+        }
+      }
+      return 0;
+    }, nullptr);
+
     bool to_init = false;
```
Comment (srinivamd) [inline lib/mcwamp.cpp:452]: LGTM

**Review by srinivamd — APPROVED** [review]
LGTM

**Comment by jeffdaily** [comment]
I'm getting a seg fault with pytorch test.sh.

**Comment by jeffdaily** [comment]
Turning CLAMP::library_path into a static function seems to avoid the seg fault.  See
https://github.com/jeffdaily/hcc/pull/1
for the diff.

**Comment by scchan** [comment]
> Turning CLAMP::library_path into a static function seems to avoid the seg fault. See
> [jeffdaily#1](https://github.com/jeffdaily/hcc/pull/1)
> for the diff.

I haven't seen any segfault... but perhaps it may have to do with the order of when the global objects get constructed.
I made a comment to your commit, could you try that?  If that works, could you submit a PR?

**Comment by scchan** [comment]
replaced by https://github.com/RadeonOpenCompute/hcc/pull/1266

**Pull request closed without merging**